### PR TITLE
round transactin total before sending it to PayPal

### PR DESCRIPTION
**Pull request by mnelson4**
## Problem this Pull Request solves

Fixes a bug where sometimes PayPal smart buttons would have an error after opening the Paypal window. Addresses https://github.com/eventespresso/eea-paypal-smart-buttons/issues/38#event-2385803346.
The problem was caused because the payment total in this case was something like 55.6000000000001, and PayPal only wants 2 decimal places. So made sure we're only sending 2 decimal places to PayPal

## How has this been tested

* [ ] Follow Josh's original steps from https://github.com/eventespresso/eea-paypal-smart-buttons/issues/38#event-2385803346
* [ ] Also try a regular payment with payPal smart buttons
* [ ] Switch your country to Japan or some other country with an unusual number of decimal places, that PayPal also accepts, and try paying. It should still work as before.

## Checklist

* [x] I have read the documentation relating to systems affected by this pull request, see https://github.com/eventespresso/event-espresso-core/tree/master/docs
* [x] User input is adequately validated and sanitized
* [x] all publicly displayed strings are internationalized (usually using `esc_html__()`, see https://codex.wordpress.org/I18n_for_WordPress_Developers)
* [x] My code is tested.
* [x] My code follows the Event Espresso code style.
* [x] My code has proper inline documentation.

### CODE REVIEW

**Comment by joshfeck** [comment]
I don't know if this was ready for testing yet, but I tested it anyway and it's still broken.

**Comment by mnelson4** [comment]
Dang, yeah it fixed the issue for me. Let me try again...

**Comment by mnelson4** [comment]
Nah I got the issue, so I don't know how it worked for me (I had thought `round` would remove the extra decimal places, but apparently it didn't.) Either way, this latest change works for me now, because it actually converts the number into a string with no precision issue.

**Comment by joshfeck** [comment]
> so I don't know how it worked for me

I think what's happening is the amount is OK if you refresh the checkout page. If you try to pay after the initial page load, it's still broken even with the latest changes on this branch.

So in order to verify this is fixed you'll probably need to work through a registration to get to the checkout page and test a payment, without refreshing the page.

**Comment by mnelson4** [comment]
> the amount is OK if you refresh the checkout page. If you try to pay after the initial page load, it's still broken even with the latest changes on this branch.

ding ding ding! Yes, although my original commit rounded the transaction total loaded by the smart buttons code; the amount passed from SPCO, which dynamically updated the amount when a promotion was applied etc, was NOT being rounded. So the latest commit does the rounding in the JS also, right before sending the amount to PayPal

**Comment by joshfeck** [comment]
The PayPal smart buttons no longer appear after pulling the latest changes to this branch. I have cleared the browser's cache just in case it's a caching issue. Here's the error in the console:

Screen Shot 2019-06-06 at 11 43 41 AM

**Comment by mnelson4** [comment]
sorry for the headache @joshfeck. It seems I neglected to add the new built file (any change to src JS files rebuilds them). So that was a waste of everyone's time, sorry!

**Comment by mnelson4** [comment]
No worries, @knazart, now you know a new way I can mess up a commit!

**Comment by joshfeck** [comment]
This works!
